aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,153 @@
+- The HTTP URI **must** end with `:batchGet`.
+- The URI path **should** represent the collection for the resource, matching
+  the collection used for simple CRUD operations. If the operation spans
+  parents, a dash (`-`) **may** be accepted as a wildcard.
+- There **must not** be a request body.
+  - If a GET request contains a body, the body **must** be ignored, and **must not** cause an error.
+- The operation **must** be atomic: it **must** fail for all resources or
+  succeed for all resources (no partial success). For situations requiring
+  partial failures, `List` (AEP-132) methods **should** be used.
+  - If the operation covers multiple locations and at least one location is
+    down, the operation **must** fail.
+
+### Request
+
+The request for a batch get method **should** be specified with the following
+pattern:
+
+{% tab proto -%}
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/{parent=publishers/*}/books:batchGet"
+  };
+}
+
+message BatchGetBooksRequest {
+  // The parent resource shared by all books being retrieved.
+  // Format: publishers/{publisher}
+  // If this is set, the parent of all of the books specified in `paths`
+  // must match this field.
+  string parent = 1 [
+    (google.api.resource_reference) = {
+      child_type: "library.com/Book"
+    }];
+
+  // The paths of the books to retrieve.
+  // A maximum of 1000 books can be retrieved in a batch.
+  // Format: publishers/{publisher}/books/{book}
+  repeated string paths = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.com/Book"
+    }];
+}
+```
+
+- The request and response schemas **must** match the method name, with `Request` and
+  `Response` suffixes.
+- A `parent` field **should** be included, unless the resource being retrieved
+  is a top-level resource, to facilitate inclusion in the URI as well to permit
+  a single permissions check. If a caller sets this field, and the parent
+  collection in the path of any resource being retrieved does not match, the
+  request **must** fail.
+  - This field **should** be required if only 1 parent per request is allowed.
+  - The field **should** identify the [resource type][aep-122-parent] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+- The request **must** include a repeated field which accepts the resource
+  paths specifying the resources to retrieve. The field **should** be named
+  `paths`.
+  - If no resource paths are provided, the API **should** error with
+    `INVALID_ARGUMENT`.
+  - The field **should** be required.
+  - The field **should** identify the [resource type][aep-122-paths] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+  - The comment for the field **should** document the maximum number of
+    paths allowed.
```

Comment:
Can this all be factored out of the proto tab into general guidance?

Reply:
It cannot, because the OAS guidance does not go in a comment field or even in the `description`, but in a `maxItems` schema attribute.

Reply:
Addressed in latest commit.